src/components/section/ArticlesSearch.jsx

```diff
@@ -0,0 +1,37 @@
+import NewsCard from "../ui/NewsCard";
+import { useSelector, useDispatch } from "react-redux";
+import { searchCategory, loadingData } from "../../features/news";
+import { useEffect } from "react";
+import { getEverything } from "../../features/api";
+const ArticlesSearch = () => {
+  const searchArticles = useSelector((state) => state.news.search);
+  const newsArticles = useSelector((state) => state.news.articles);
+  const inputValue = useSelector((state) => state.news.inputValue);
+  const dispatch = useDispatch();
+
+  const filterAllArticles = [
+    ...(searchArticles?.articles || []),
+    ...(newsArticles?.articles || []),
+  ]?.filter(
+    (article) =>
+      article.title.toLowerCase().includes(inputValue.toLowerCase()) ||
+      article.source.name.toLowerCase().includes(inputValue.toLowerCase())
+  );
+  useEffect(() => {
```

Comment:
This will make uneneccary fetch, since inptValue will always change as the user time in the input field, it means it will fetch with every character